packages/datagovmy-ui/src/data-catalogue/Main/index.tsx

```diff
@@ -43,18 +43,27 @@ interface CatalogueIndexProps {
   site: SiteName;
 }
 
+const getSourceTranslation = (
+  source: string,
+  language: keyof (typeof SOURCE_TRANSLATIONS)[string]
+) => {
+  return SOURCE_TRANSLATIONS[source]?.[language] || source;
+};
+
 const CatalogueIndex: FunctionComponent<CatalogueIndexProps> = ({
   query,
   collection,
   sources,
   site,
 }) => {
-  const { t } = useTranslation(["catalogue", "common"]);
+  const { t, i18n } = useTranslation(["catalogue", "common"]);
   const scrollRef = useRef<Record<string, HTMLElement | null>>({});
   const filterRef = useRef<CatalogueFilterRef>(null);
   const { size } = useContext(WindowContext);
+  const language = i18n.language; // Get the current language
+
   const sourceOptions = sources.map(source => ({
-    label: source,
+    label: getSourceTranslation(source, language as "ms-MY" | "en-GB"),
```

Comment:
can probably do this instead, `label: t(`agencies:${source.toLowerCase()}.full`),`

Reply:
Ok